pkg/test/framework/components/istio/operator.go

```diff
@@ -85,6 +89,12 @@ func (i *operatorComponent) Close() (err error) {
 			if e := cluster.DeleteContents("", removeCRDs(i.installManifest[cluster.Name()])); e != nil {
 				err = multierror.Append(err, e)
 			}
+			if e := cluster.DeleteNamespace(i.settings.SystemNamespace); e != nil {
```

Comment:
I don't think you should need this ... this should already be done elsewhere. Where you not seeing this happen?

Reply:
`*kubeNamespace.Close()` calls it, but `*Accessor.DeleteNamespace` deletes in the foreground without waiting. When the next suite starts the secrets are still around and istio has issues deploying. 

An alternative to nuking the entire NS would be just deleting the secrets?

Reply:
> but *Accessor.DeleteNamespace deletes in the foreground

Do you mean that it deletes in the background? Perhaps we can pass options to `DeleteNamespace` for force it to wait?

Reply:
I thought the `propagationPolicy=foreground` might be why it returns instantly. Doesn't look like there is a Wait option in the api, but I'll move the WaitForNamespaceDeletion to the Namespace cleanup instead of here.

Reply:
After my last commit I'm thinking the `*kubeNamespace.Close` method isn't getting called at all. When the second cluster starts to deploy it fails to create the namespace and push `cacerts` secrets, then hangs on `Processing resources for istiod` when deploying the master. 

<details>
<summary>Logs</summary>

```
2020-05-01T20:30:51.007212Z     info    ci      ================================
2020-05-01T20:30:52.698577Z     info    ci      failed creating namespace istio-system on cluster cluster-0. This can happen when deploying multiple control planes. Error: namespaces "istio-system" alread
y exists
2020-05-01T20:30:52.706046Z     info    ci      failed to create CA secrets on cluster cluster-0. This can happen when deploying multiple control planes. Error: secrets "cacerts" already exists
2020-05-01T20:30:53.159673Z     info    ci      failed creating namespace istio-system on cluster cluster-1. This can happen when deploying multiple control planes. Error: namespaces "istio-system" alread
y exists
2020-05-01T20:30:53.166909Z     info    ci      failed to create CA secrets on cluster cluster-1. This can happen when deploying multiple control planes. Error: secrets "cacerts" already exists
2020-05-01T20:30:53.484159Z     info    ci      Running istio control plane on cluster cluster-0 [manifest apply --skip-confirmation --wait -f /home/prow/go/src/istio.io/istio/tests/integration/iop-integr
ation-test-defaults.yaml -f /logs/artifacts/multicluster/remote-3ebde6cc5dd/_suite_context/istio-deployment-716369742/iop.yaml --set values.global.imagePullPolicy=IfNotPresent --charts /home/prow/go/src/i
stio.io/istio/manifests --set values.global.hub=istio-testing --set values.global.tag=istio-testing --set values.global.imagePullPolicy=IfNotPresent --set values.global.multiCluster.clusterName=cluster-0
--set values.global.meshExpansion.enabled=true]

- Processing resources for Istio core.
✔ Istio core installed
- Processing resources for Istiod.

```

</details>

Reply:
Looks like it creates the namespace using the Acessor so it doesn't get tracked in the resource.Context.

Reply:
similar problem in an existing test: https://github.com/istio/istio/blob/b55c55b41210e05ad8e53c795a856998e047383f/tests/integration/security/util/cert/cert.go#L110

I don't think we delete the namespace as part of cleanup anymore because its slow

Reply:
So are we more in favor of:
1. cleanup istio-system
2. cleanup secrets in istio-system 
3. if (exists) { delete ns }

Reply:
@stevenctl @howardjohn 

I think the *right* thing to do is to generally never use `istio-system` in tests, but rather have each environment use a generated system namespace (e.g. `istio-system-12345`), similarly to what we do for other namespaces currently.  Then we would be able to delete the namespace at the end without having to wait.

This would also conceivably allow us to run multiple tests in parallel without having them step on each other (except for global resources, like crds).

What do you guys think?

Reply:
> Then we would be able to delete the namespace at the end without having to wait.

We don't cleanup asynchronously in general but I think we can for namespace since we will trigger the delete then k8s will handle the rest. That would probably work well.

> This would also conceivably allow us to run multiple tests in parallel without having them step on each other (except for global resources, like crds).

This was one thought we had with the multi revision. It might work - definitely worth an experiment. Probably outside the scope of this PR though.

WRT using not istio-system, I agree. We should probably improve our testing of not istio-system setups (just found a bug around this last week actually), this would be a good win for that as well. `i.settings.SystemNamespace` is meant to do that in many ways, but isn't fully hooked up to randomize a ns per test, just statically changing istio-system to another constant.

For this PR, I would suggest we cleanup secrets then in a followup see if we can do randomized namespaces?

Reply:
sgtm

Reply:
I think something is hanging out other than secrets. Even deleting secrets with propagationPolicy=Foreground, the second istio deploy times out.

Reply:
@stevenctl I assume this is causing the multi-cluster tests to break? Maybe we delete the NS manually only if it's a multi-cluster test for now? Then we can follow-up with the real fix in a follow-on PR?

Reply:
sounds good; working on the real fix in parallel, just a bit of hairiness because the namespace package depends on this istio package.

Reply:
Ah right, I forgot about that ugly bit. Ideally it shouldn't. Feel free to work around it or "fix" that dependency ... I'm fine either way.